crates/metrics/src/config.rs

```diff
@@ -0,0 +1,113 @@
+use once_cell::sync::Lazy;
+use strum::IntoEnumIterator;
+use strum_macros::{
+    Display,
+    EnumIter,
+    EnumString,
+};
+
+#[derive(Debug, Display, Clone, Copy, PartialEq, EnumString, EnumIter)]
+#[strum(serialize_all = "lowercase")]
+pub enum Module {
+    Importer,
+    P2P,
+    Producer,
+    TxPool, /* TODO[RC]: Not used. Add support in https://github.com/FuelLabs/fuel-core/pull/2321 */
+    GraphQL, // TODO[RC]: Not used... yet.
+}
+
+#[derive(Debug, Clone, Default)]
+pub struct Config(Vec<Module>);
```

Comment:
nit: If we are not iterating and don't have duplicate it might be worth having an hashset

Reply:
I don't have a proof at hand, but I remember benchmarks showing that `Vec` is usually faster due to cache-locality, especially for small sets.

Unless you have something else than performance in mind?

Reply:
Trusting you, in any case it's not much.